### PR TITLE
Bulkapi v2

### PR DESCRIPTION
**Pull request by Tharsanan1**

### CODE REVIEW

File: src/main/java/org/wso2/carbon/esb/connector/exception/InvalidConfigurationException.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2022, WSO2 LLC. (http://www.wso2.com).
```
Comment (sajinieKavindya) [inline src/main/java/org/wso2/carbon/esb/connector/exception/InvalidConfigurationException.java:2]: Shall we update the license headers? 2023

File: src/main/java/org/wso2/carbon/esb/connector/utils/JobType.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package org.wso2.carbon.esb.connector.utils;
+
+public enum JobType {
```
Comment (SanojPunchihewa) [inline src/main/java/org/wso2/carbon/esb/connector/utils/JobType.java:3]: Do we have usage of this enum?

File: src/main/java/org/wso2/carbon/esb/connector/utils/SalesforceUtils.java [hunk 3]
```diff
@@ -0,0 +1,432 @@
...
+
+
+    /**
+     * Create an OMElement.
+     *
+     * @param elementName Name of the element
+     * @param value       Value to be added
+     * @return OMElement or null if error
+     */
+    public static OMElement createOMElement(String elementName, String value) {
```
Comment (SanojPunchihewa) [inline src/main/java/org/wso2/carbon/esb/connector/utils/SalesforceUtils.java:305]: Shall we remove these unused methods?

File: src/main/java/org/wso2/carbon/esb/connector/requests/SalesforceRequest.java [hunk 4]
```diff
@@ -0,0 +1,362 @@
...
+        headers.put(RequestConstants.HTTP_HEADER_CONTENT_TYPE, RequestConstants.APPLICATION_JSON);
+        headers.put(RequestConstants.HTTP_HEADER_ACCEPT, RequestConstants.APPLICATION_JSON);
+        headers.put(RequestConstants.X_PRETTY_PRINT, "1");
+        RestRequest restRequest =
+                new RestRequest(HttpMethod.GET, SalesforceUtils.getGetJobFailedResultsUrl(salesforceConfig, jobId),
+                        null, headers);
+        return sendRequest(restRequest).getResponse();
+    }
+
+    public void getJobSuccessfulResultsAndStoreInFile(String jobId, String filePath) throws SalesforceConnectionException, InvalidConfigurationException {
```
Comment (SanojPunchihewa) [inline src/main/java/org/wso2/carbon/esb/connector/requests/SalesforceRequest.java:264]: This seems to exceed the maximum line length. There we a few other places as well. Shall we format the code?

File: src/main/java/org/wso2/carbon/esb/connector/utils/JobType.java (hunk 2 above)
Comment (Tharsanan1) [reply src/main/java/org/wso2/carbon/esb/connector/utils/JobType.java]: removed the unused enums.

File: src/main/java/org/wso2/carbon/esb/connector/utils/SalesforceUtils.java (hunk 3 above)
Comment (Tharsanan1) [reply src/main/java/org/wso2/carbon/esb/connector/utils/SalesforceUtils.java]: removed the unused methods.

File: src/main/java/org/wso2/carbon/esb/connector/requests/SalesforceRequest.java (hunk 4 above)
Comment (Tharsanan1) [reply src/main/java/org/wso2/carbon/esb/connector/requests/SalesforceRequest.java]: Done

**Review by rosensilva — APPROVED** [review]

**Pull request merged**